### ISSUE TITLE
Grammar missing ability to have just a function call.

### ISSUE THREAD

**Issue by dj2** [gpuweb/gpuweb#716]
Currently there is no way in the grammar to have a function call by itself on a line, it is not in the list of `statements`.

For example, the following will fail to parse:
```
fn a() -> void {
  return;
}

fn b() -> void {
  a();
  return;
}
```

The `statements` list needs to be extended to have something like:

```
 | IDENT paren_rhs_stmt
```

This means we'll have to do lookahead at in the statement parsing because you can't tell if this is a call or an assignment straight away.

**Comment by othermaciej** [comment]
Clearly a bare function call statement should be allowed. But shouldn't it be?

I think the function call statement would need to be something like `singular_expression PAREN_LEFT argument_expression_list* PAREN_RIGHT`? 

`paren_rhs_stmt` would only express exactly one function argument, rather than zero or more than one, and `IDENT` would only allow `foo(bar)`, not `foo.bar(baz)`, which is allowed to appear on the right hand side of an assignment.

**Comment by dj2** [comment]
We don't have function pointers, so you can't do `foo.bar(baz)`.  Looking at this again, we probably want something like:

```
(IDENT NAMESPACE)* IDENT PAREN_LEFT argument_expression_list? PAREN_RIGHT
```

This would allow local function calls and namespaced function calls.

**Comment by othermaciej** [comment]
Perhaps it's a bug that the current grammar allows `foo.bar(baz)` on the right side of an assignment (at least as I read it?)

**Comment by dj2** [comment]
You can have a function call on the right side of an assignment, the issue with is that the function call is called off of the member of the struct, which we should disallow, but I'm not sure if that should be done in the grammar or in validation.

We'd have the same issue if you did something like:

```
var a : vec3<f32>;
a[0]();
```

which is also allowed by the grammar but should be either fixed in the grammar or a validation error. I'm not sure how hard it will be to disallow in the grammar. Possibly we just need to pull the `()` statement out of `postfix_expression` and put it somewhere else?

**Comment by othermaciej** [comment]
I think these things should be fixed in the grammar and are worth a separate issue.

**Comment by dj2** [comment]
Sure, feel free to file a new issue. If you've got thoughts on grammar changes please add those as well.

**Comment by dj2** [comment]
This was fixed by #757. If there are further restrictions please file new issues.
